src/libstd/collections/hash/map.rs

```diff
@@ -1273,6 +1273,13 @@ impl<'a, K, V> Clone for Iter<'a, K, V> {
     }
 }
 
+#[stable(feature = "std-debug", since = "")]
+impl<'a, K, V> fmt::Debug for Iter<'a, K, V> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.pad("Iter {{ .. }}")
```

Comment:
I don't think redundant bounds are bad, there's good reason for it.

Displaying the elements is simple using `.entries()` https://doc.rust-lang.org/nightly/std/fmt/struct.DebugMap.html